pages/overview/[id].tsx

```diff
@@ -0,0 +1,202 @@
+import type { NextPage } from 'next';
+import { ethers } from 'ethers';
+import { useEffect, useState, useContext } from 'react';
+import { useRouter } from 'next/router';
+import AddressError from '../../components/overview/AddressError';
+import { AssetsContext } from '../../contexts/AssetsContext';
+import fetchReceivedAssets from '../../utils/fetchReceivedAssets';
+import useEthersProvider from '../../hooks/useEthersProvider';
+import { ethersProvider } from '../../types';
+import isLSP7orLSP8 from '../../utils/isLSP7orLSP8';
+import fetchLSP4Metadata from '../../utils/fetchLSP4Metadata';
+import fetchLSP7Balance from '../../utils/fetchLSP7Balance';
+import fetchLSP8Metadata from '../../utils/fetchLSP8Metadata';
+import fetchLSP8TokensIds from '../../utils/fetchLSP8TokensIds';
+import isUP from '../../utils/isUP';
+import useWeb3Provider from '../../hooks/useWeb3Provider';
+import { Lsp7AssetType, Lsp8AssetType } from '../../contexts/AssetsContext';
+import LSP7Table from '../../components/overview/LSP7sTable';
+import LSP8Table from '../../components/overview/LSP8Table';
+import { luksoImg } from '../../constants';
+
+const AdressOverview: NextPage = () => {
+  const router = useRouter();
+  const [addressError, setAddressError] = useState<string>('');
+  const [isUniversalProfile, setIsUniversalProfile] = useState<boolean>(false);
+  const [loading, setLoading] = useState<boolean>(false);
+
+  const [showLSP7, setShowLSP7] = useState<boolean>(true);
+
+  const { lsp7Assets, setLsp7Assets, lsp8Assets, setLsp8Assets } =
+    useContext(AssetsContext);
+  const provider = useEthersProvider() as ethersProvider;
+
+  //ERC725 does not support ethers provider
+  const web3Provider = useWeb3Provider();
+
+  const getAssets = async (address: string) => {
```

Comment:
This is a very big function which makes it difficult to read. I think we can split it down into smaller reusable funtions

Reply:
I agree - also, these functions can be moved into a `services` folder where you put all these functions. Then you don't have to import so many things in this component.